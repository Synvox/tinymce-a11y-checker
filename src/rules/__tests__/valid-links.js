import rule from "../valid-links"

let body, a

beforeEach(() => {
  body = document.createElement("body")
  a = document.createElement("a")
  body.appendChild(a)
  a.textContent = "Link Text"
})

describe("test", () => {
  test("returns true if not A element", () => {
    expect(rule.test(document.createElement("div"))).toBe(true)
  })

  test("returns true for valid links", () => {
    a.setAttribute("href", "http://example.com")
    expect(rule.test(a)).toBe(true)
    a.setAttribute("href", "mailto:user@example.com")
    expect(rule.test(a)).toBe(true)
  })

  test("returns false for invalid links", () => {
    a.setAttribute("href", "http://something weird")
    expect(rule.test(a)).toBe(false)
    a.setAttribute("href", "plaintext")
    expect(rule.test(a)).toBe(false)
  })
})

describe("data", () => {
  test("returns the proper object", () => {
    expect(rule.data(a)).toMatchSnapshot()
  })
})

describe("form", () => {
  test("returns the proper object", () => {
    expect(rule.form(a)).toMatchSnapshot()
  })
})

describe("update", () => {
  test("returns same element", () => {
    expect(rule.update(a, {})).toBe(a)
  })
  test("does not change href if href does not change", () => {
    const href = "http://example.com"
    a.setAttribute("href", href)
    expect(rule.update(a, { href })).toBe(a)
  })
  test("changes href if it has been changed", () => {
    const href = "bad"
    a.setAttribute("href", href)
    expect(rule.update(a, { href: "http://good.com" })).toBe(a)
  })
})

describe("rootNode", () => {
  test("returns the parentNode of an element", () => {
    expect(rule.rootNode(a).tagName).toBe("BODY")
  })
})

describe("message", () => {
  test("returns the proper message", () => {
    expect(rule.message()).toMatchSnapshot()
  })
})

describe("why", () => {
  test("returns the proper why message", () => {
    expect(rule.why()).toMatchSnapshot()
  })
})
