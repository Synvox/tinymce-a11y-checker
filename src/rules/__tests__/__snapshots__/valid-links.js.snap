// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data returns the proper object 1`] = `
Object {
  "href": null,
}
`;

exports[`form returns the proper object 1`] = `
Array [
  Object {
    "dataKey": "href",
    "label": "Change the URL for this link",
  },
]
`;

exports[`message returns the proper message 1`] = `"Links should be valid."`;

exports[`why returns the proper why message 1`] = `"Invalid links become broken links which are confusing to all users."`;
